sdk/apimanagement/Azure.ResourceManager.ApiManagement/tests/Scenario/ApiManagementServiceCollectionTests.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.ResourceManager.ApiManagement.Models;
+using Azure.ResourceManager.Models;
+using Azure.ResourceManager.Network;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.ApiManagement.Tests
+{
+    public class ApiManagementServiceCollectionTests : ApiManagementManagementTestBase
+    {
+        public ApiManagementServiceCollectionTests(bool isAsync)
+                    : base(isAsync)//, RecordedTestMode.Record)
+        {
+        }
+
+        private VirtualNetworkCollection VNetCollection { get; set; }
+
+        private async Task<ApiManagementServiceCollection> GetApiManagementServiceCollectionAsync()
+        {
+            var resourceGroup = await CreateResourceGroupAsync();
+            VNetCollection = resourceGroup.GetVirtualNetworks();
+            return resourceGroup.GetApiManagementServices();
+        }
+
+        [Test]
+        public async Task CRUD()
+        {
+            var collection = await GetApiManagementServiceCollectionAsync();
+            var apiName = Recording.GenerateAssetName("testapi-");
+            var data = new ApiManagementServiceData(AzureLocation.EastUS, new ApiManagementServiceSkuProperties(ApiManagementServiceSkuType.Developer, 1), "Sample@Sample.com", "sample")
+            {
+                Identity = new ManagedServiceIdentity(ManagedServiceIdentityType.SystemAssigned)
+            };
+            var apiManagementService = (await collection.CreateOrUpdateAsync(WaitUntil.Completed, apiName, data)).Value;
+            Assert.AreEqual(apiManagementService.Data.Name, apiName);
+
+            // TagOperation
+            await apiManagementService.AddTagAsync("testkey", "testvalue");
+            apiManagementService = (await apiManagementService.GetAsync()).Value;
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Key, "testkey");
+            Assert.AreEqual(apiManagementService.Data.Tags.FirstOrDefault().Value, "testvalue");
+
```

Comment:
This assertion is fragile. Could you please change this (and all the ocurrence like this) to something  like
```
Assert.IsTrue(apiManagementService.Data.Tags.ContainsKey("testkey"));
Assert.AreEqual(apiManagementService.Data.Tags["testkey"], "testvalue");
```
The first assertion is actually optional. if the key does not exist, accessing it will give you an excpetion.